### ISSUE TITLE
pydantic validation with response_model have bug

### ISSUE THREAD

**Issue by shurshilov** [tiangolo/fastapi#5061]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter
from typing import Union
from pydantic import BaseModel, Field, root_validator, parse_obj_as, ValidationError
router = APIRouter(tags=["http"])
from ..test.data import CLIENTS, TARGETS
class Client(BaseModel):
    id: int = Field(..., title="Id")
    short_name_loc: Union[str, None] = Field(..., title="short_name_loc")
    long_name_loc: Union[str, None] = Field(..., title="long_name_loc")

@router.get("/api/clients/", response_model=list[Client])
def clients(item: list[Client]):
    c = CLIENTS
    b:list[Client] = parse_obj_as(list[Client], c)
    return b
```

### Description

 Why if I delete arg in clients func item: list[Client] and pydantic model invalid it got500 error
 If i added item: list[Client] then I got pretty json error in browser?
 I think args dont related with output response

### Operating System

Windows

### FastAPI Version

0.68.2

### Python Version

3.9.1

**Comment by shurshilov** [comment]
Also analogy 
@router.get("/api/backend/version", response_model=str)
def version(item: str = ''):
    return {}
give 500 error BUT expected beauty pydantic error

**Comment by zoliknemet** [comment]
@shurshilov 
Few comments on your code:
1. HTTP GET is not suppose to have request body - your method should be like this: 
```python
@router.get("/api/clients/", response_model=list[Client])
def clients():
    return CLIENTS
```

2. when `response_model` is defined it always tries to instantiate it - in this case it will take CLIENTS and make object of type List[Client] even if returned CLIENTS is dict it will work

3. You get 500 because your Client takes ID as INTEGER - your data says
```
CLIENTS = [{"id": "1a", "short_name_loc": "Demo", "long_name_loc": "DEMO"}]
```
here in dict, ID is String and can not map it to Client - ValidationError is raised

4. In second comment you returned DICTIONARY and response_model is STRING - you have to have good types declared and returned

If I understood correctly you have issue with 500 error instead you expect 400 to be returned.
I would say that 500 is valid in this case, it is internal error, code has bugs.
400 is used if user called your API with wrong parameters/data - 400 Bad Request

**Comment by shurshilov** [comment]
Thank you for answer
May be best way send not 500 internal server with tracelog from pydantic? its way use in different frameworks...

**Comment by zoliknemet** [comment]
You can always catch it with exception handler

```python
@app.exception_handler(Exception)
async def http_exception_handler(request, exc):
    return PlainTextResponse(traceback.format_exc())

```

**Comment by shurshilov** [comment]
> @zoliknemet

Yes, just strange that for 400 response set Text error but for 500 not =\
